weather.go

```diff
@@ -1,5 +1,138 @@
 package weather
 
-func Get(location, token string) ([]byte, error) {
-	return nil, nil
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math"
+	"net/http"
+	"os"
+	"strings"
+)
+
+type Weather struct {
+	Name    string
+	Weather []struct {
+		ID          int
+		Main        string
+		Description string
+		Icon        string
+	}
+	Main struct {
+		Temp      float64
+		FeelsLike float64
+		TempMin   float64
+		TempMax   float64
+		Humidity  int
+	}
+}
+
+type CityUnknown struct {
+	Cod     string
+	Message string
+}
+
+type Conditions struct {
+	City    string
+	OneWord string
+	Celsius float64
+}
+
+func CliOutput(token string, args []string) (output string) {
+
+	location := LocationFromArgs(args)
+	url := BuildURL(token, location)
+	callURL := CallURL(url)
+	conditions, err := Get(callURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	output = fmt.Sprintf("city: %s\nweather: %s\nCelsius: %v\n", conditions.City, conditions.OneWord, conditions.Celsius)
+
+	return output
+}
+
+func LocationFromArgs(args []string) string {
+
+	if len(os.Args) == 1 {
+		fmt.Fprintf(os.Stderr, "please set a location e.g. london\n")
```

Comment:
It's a good idea to reserve the power to write to the terminal to just one function: `RunCLI` or the equivalent (in this program, it's called `CliOutput`). All other functions should return error instead, and then `RunCLI` can decide what message to print about them.